peg_src/jsdoctype.pegjs

```diff
@@ -980,14 +981,17 @@ RecordTypeExprEntries = first:RecordTypeExprEntry restWithComma:(_ "," _ RecordT
                         }, [first]);
                       }
 
-
-RecordTypeExprEntry = keyInfo:RecordTypeExprEntryKey _ ":" _ value:RecordTypeExprEntryOperand {
+RecordTypeExprEntry = keyInfo:RecordTypeExprEntryKey _ optional:"?"? _ ":" _ value:RecordTypeExprEntryOperand {
```

Comment:
Also, this currently allows `{optional?: type=}`, which is invalid.

To fix this, you’ll probably have to separate this into two branches where the one with `?` doesn’t allow `=`‑based optionalness, and the other branch does.

Reply:
I understand what you mean, but do we actually have to disallow such a duplicate syntax? I would say yes if the parsing was meant as a validator, but as the two intents aren't conflicting, is there any actual harm in actually "allowing" it?

Reply:
¯\\\_(ツ)\_/¯

Reply:
I vote for the simpler code. Linters can easily write a check for it if there isn't a huge demand for it, and their code will be simpler since it operates on a complete tree.